presidio-analyzer/presidio_analyzer/predefined_recognizers/date_recognizer.py

```diff
@@ -48,15 +48,30 @@ class DateRecognizer(PatternRecognizer):
             0.6,
         ),
         Pattern(
-            "dd.mm.yyyy",
-            r"\b(([1-9]|0[1-9]|[1-2][0-9]|3[0-1])\.([1-9]|0[1-9]|1[0-2])\.\d{4})\b",
+            "dd.mm.yyyy or dd.mm.yy",
+            r"\b(([1-9]|0[1-9]|[1-2][0-9]|3[0-1])\.([1-9]|0[1-9]|1[0-2])\.(\d{4}|\d{2}))\b",  # noqa: E501
             0.6,
         ),
         Pattern(
-            "dd-MMM-yyyy",
-            r"\b(([1-9]|0[1-9]|[1-2][0-9]|3[0-1])-(JAN|FEB|MAR|APR|MAY|JUN|JUL|AUG|SEP|OCT|NOV|DEC)-\d{4})\b",  # noqa: E501
+            "dd-MMM-yyyy or dd-MMM-yy",
+            r"\b(([1-9]|0[1-9]|[1-2][0-9]|3[0-1])-(JAN|FEB|MAR|APR|MAY|JUN|JUL|AUG|SEP|OCT|NOV|DEC)-(\d{4}|\d{2}))\b",  # noqa: E501
             0.6,
         ),
+        Pattern(
+            "MMM-yyyy or MMM-yy",
+            r"\b((JAN|FEB|MAR|APR|MAY|JUN|JUL|AUG|SEP|OCT|NOV|DEC)-(\d{4}|\d{2}))\b",  # noqa: E501
+            0.6,
+        ),
+        Pattern(
+            "dd-MMM or dd-MMM",
+            r"\b(([1-9]|0[1-9]|[1-2][0-9]|3[0-1])-(JAN|FEB|MAR|APR|MAY|JUN|JUL|AUG|SEP|OCT|NOV|DEC))\b",  # noqa: E501
+            0.6,
+        ),
+        Pattern(
+            "mm/yyyy or mm/yy",
```

Comment:
Should this be mm/yyyy or mm/yy or m/yyyy or m/yy?

Reply:
Good point

Reply:
1. Agree
2. Agree as well, but this would mean splitting into two Patterns.
3. Not working for beginning and end of sentence . For example starting an input or text with just date, or ending a sentence with a date.

Reply:
@SharonHart for (3) this sounds like a good approach. I would actually add it to all.

Reply:
Actually the \b word boundary catches all of those cases, thought is won't. So nothing needs to be done for (3)

Reply:
Oh ok! my bad...